provider/pkg/clients/unstructured.go

```diff
@@ -65,22 +65,35 @@ func ToUnstructured(object metav1.Object) (*unstructured.Unstructured, error) {
 // This process normalizes semantically-equivalent resources into an identical output, which is important for diffing.
 // If the scheme is not defined, then return the original resource.
 func Normalize(uns *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	var result *unstructured.Unstructured
+
 	if IsCRD(uns) {
-		return normalizeCRD(uns), nil
+		result = normalizeCRD(uns)
+	} else if IsSecret(uns) {
+		result = normalizeSecret(uns)
+	} else {
+		obj, err := FromUnstructured(uns)
+		// Return the input resource rather than an error if this operation fails.
+		if err != nil {
+			return uns, nil
+		}
+		result, err = ToUnstructured(obj)
+		// Return the input resource rather than an error if this operation fails.
+		if err != nil {
+			return uns, err
```

Comment:
Was it intentional to return the `err` here? The comment seems to say otherwise.